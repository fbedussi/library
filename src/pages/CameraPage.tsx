import 'react-html5-camera-photo/build/css/index.css'

import React from 'react'
import Camera, { FACING_MODES, IMAGE_TYPES } from 'react-html5-camera-photo'
import { useDispatch } from 'react-redux'
import { useNavigate } from 'react-router-dom'

import BackLink from '../components/BackLink'
import { PageWrapper, ToolbarStyled, TopAppBar } from '../components/CommonComponents'
import { TDispatch } from '../model/types'
import photosActions from '../store/photos/actions'

const CameraPage: React.FC = () => {
	const dispatch: TDispatch = useDispatch();

	const navigate = useNavigate();

	return (
		<PageWrapper>
			<TopAppBar position="fixed" color="primary">
				<ToolbarStyled>
					<BackLink />
				</ToolbarStyled>
			</TopAppBar>
			<Camera
				onTakePhoto={dataUri => {
					dispatch(photosActions.uploadPhoto(dataUri)).then(() =>
						navigate('/add'),
					);
				}}
				idealFacingMode={FACING_MODES.ENVIRONMENT}
				imageType={IMAGE_TYPES.JPG}
				imageCompression={0.97}
				isFullscreen={false}
				isImageMirror={false}
			/>
		</PageWrapper>
	);
};

export default CameraPage;
