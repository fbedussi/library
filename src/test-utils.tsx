import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'

import {
  ByRoleMatcher, ByRoleOptions, render as rtlRender,
  screen as rtlScreen
} from '@testing-library/react'

import { RootState } from './model/model'
import { getTestStore } from './store'

function render(
	ui: JSX.Element,
	{
		initialState = {},
		store = getTestStore(initialState),
		dispatch,
		route = '/',
		...renderOptions
	} = {} as {
		initialState?: Partial<RootState>;
		store?: any;
		dispatch?: any;
		route?: string;
	},
) {
	const Wrapper: React.FC<{ children?: JSX.Element }> = ({ children }) => {
		return (
			<Provider store={dispatch ? { ...store, dispatch } : store}>
				<BrowserRouter>{children}</BrowserRouter>
			</Provider>
		);
	};
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const screen = {
	...rtlScreen,
	getByRole: <T extends HTMLElement>(
		text: ByRoleMatcher,
		options?: ByRoleOptions | undefined,
		waitForElementOptions?: unknown,
	) => rtlScreen.getByRole(text, options, waitForElementOptions) as T,
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { render, screen };
