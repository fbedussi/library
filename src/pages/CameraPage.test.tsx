import React from 'react'

import { render } from '../test-utils'
import CameraPage from './CameraPage'

jest.mock('../data', () => ({}));

describe('CameraPage', () => {
	it('renders correcly', () => {
		const page = render(<CameraPage />);
		expect(page).toMatchSnapshot();
	});
});
