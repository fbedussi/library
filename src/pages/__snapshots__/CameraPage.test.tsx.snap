// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraPage renders correcly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ eJvVid"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary sc-jSgupP cfyQgS mui-fixed MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular sc-gKsewC bvkgnn MuiToolbar-gutters"
          >
            <button
              aria-label="open drawer"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </header>
        <div
          class="react-html5-camera-photo "
        >
          <div
            class=" normal"
            id="white-flash"
          />
          <img
            alt="camera"
            src=""
            style="display: none;"
          />
          <video
            autoplay=""
            playsinline=""
            style="transform: none; display: inline-block;"
          />
          <div
            id="container-circles"
          >
            <div
              id="outer-circle"
            >
              <div
                class=""
                id="inner-circle"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ eJvVid"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary sc-jSgupP cfyQgS mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular sc-gKsewC bvkgnn MuiToolbar-gutters"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
      <div
        class="react-html5-camera-photo "
      >
        <div
          class=" normal"
          id="white-flash"
        />
        <img
          alt="camera"
          src=""
          style="display: none;"
        />
        <video
          autoplay=""
          playsinline=""
          style="transform: none; display: inline-block;"
        />
        <div
          id="container-circles"
        >
          <div
            id="outer-circle"
          >
            <div
              class=""
              id="inner-circle"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
